service/src/main/kotlin/fi/espoo/evaka/jamix/JamixService.kt

```diff
@@ -104,32 +103,80 @@ class JamixService(
 
     fun syncDiets(db: Database.Connection, clock: EvakaClock, job: AsyncJob.SyncJamixDiets) {
         if (client == null) error("Cannot sync diet list: JamixEnv is not configured")
-        val warningEmails =
-            fetchAndUpdateJamixDiets(client, db, clock, emailEnv.sender(Language.fi))
-        try {
-            fetchAndUpdateJamixTextures(client, db)
-        } catch (e: Exception) {
-            logger.error(e) { "Failed to sync meal textures from Jamix" }
-        } finally {
-            warningEmails.forEach { email ->
-                try {
-                    emailClient.send(email)
-                } catch (e: Exception) {
-                    logger.error(e) { "Failed to send warning email" }
-                }
-            }
+
+        val now = clock.now()
+        val nulledChildren = fetchAndUpdateJamixDiets(client, db, now.toLocalDate())
+        if (nulledChildren.isNotEmpty()) {
+            logger.warn(
+                "Jamix diet list update caused ${nulledChildren.size} child special diets to be set to null"
+            )
+            scheduleSpecialDietNullificationWarningEmails(db, now, nulledChildren)
```

Comment:
I think this need to be part of the same transaction as fetchAndUpdateJamixDiets or otherwise diets may be reset without scheduling succeeding

Reply:
Fixed